### PR TITLE
feat(severities): add severities endpoint part 1

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

First step of severities endpoint (archi + unit tests)

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Later on

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Severity/RealTime/Application/UseCase/FindSeverity.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Severity\RealTime\Application\UseCase;
```
Comment (jeremyjaouen) [inline src/Core/Severity/RealTime/Application/UseCase/FindSeverity.php:24]:
```suggestion
namespace Core\Severity\RealTime\Application\UseCase\FindSeverity;
```
each UseCase should have its own folder too

**Review by jeremyjaouen — APPROVED** [review]

File: tests/php/Core/Severity/RealTime/Domain/SeverityTest.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Tests\Core\Severity\RealTime\Domain\Model;
```
Comment (kduret) [inline tests/php/Core/Severity/RealTime/Domain/SeverityTest.php:23]: can you move this test in a `Model` directory

File: tests/php/Core/Severity/RealTime/Application/UseCase/FindSeverityTest.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Core\Severity\RealTime\Application\UseCase;
```
Comment (kduret) [inline tests/php/Core/Severity/RealTime/Application/UseCase/FindSeverityTest.php:24]: Can you move this test in `FindSeverity` directory
```suggestion
namespace Tests\Core\Severity\RealTime\Application\UseCase\FindSeverity;
```

**Review by TamazC — APPROVED** [review]

File: tests/php/Core/Severity/RealTime/Application/UseCase/FindSeverityTest.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        ->method('findAll')
+        ->willThrowException(new \Exception());
+
+    $presenter = new FindSeverityPresenterStub($this->presenterFormatter);
+    $useCase($presenter);
+
+    expect($presenter->getResponseStatus())->toBeInstanceOf(ErrorResponse::class);
+    expect($presenter->getResponseStatus()?->getMessage())->toBe(
+        'An error occured while retrieving severities'
+    );
```
[the remark this answers is not in the document]
Comment (TamazC) [reply tests/php/Core/Severity/RealTime/Application/UseCase/FindSeverityTest.php:52]: chaining

File: tests/php/Core/Severity/RealTime/Application/UseCase/FindSeverityTest.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    $severity = new Severity(1, 'name', 50, $icon);
+
+    $this->repository
+        ->expects($this->once())
+        ->method('findAll')
+        ->willReturn([$severity]);
+
+    $presenter = new FindSeverityPresenterStub($this->presenterFormatter);
+    $useCase($presenter);
+    expect($presenter->response)->toBeInstanceOf(FindSeverityResponse::class);
```
[the remark this answers is not in the document]
Comment (TamazC) [reply tests/php/Core/Severity/RealTime/Application/UseCase/FindSeverityTest.php:71]: chaining

**Review by kduret — APPROVED** [review]

**Review by TamazC — APPROVED** [review]

**Pull request merged**